test/Autofac.Integration.WebApi.Owin.Test/DependencyScopeHandlerFixture.cs

```diff
@@ -64,14 +64,76 @@ public async void RemoveAutofacDependencyScopeFromHttpRequestMessageAfterLeaving
         Assert.DoesNotContain(HttpPropertyKeys.DependencyScope, request.Properties);
     }
 
+    [Fact]
+    public void RemoveAutofacDependencyScopeAfterTaskExecutes()
+    {
+        // Test using Task semantics rather than async/await to ensure both work.
+        var request = new HttpRequestMessage();
+        var context = new OwinContext();
+        request.Properties.Add("MS_OwinContext", context);
+
+        var container = new ContainerBuilder().Build();
+        context.Set(Constants.OwinLifetimeScopeKey, container);
+
+        var flag = new AutoResetEvent(false);
+        var fakeHandler = new FakeInnerHandler(_ =>
+        {
+            return Task.Factory.StartNew(() =>
+            {
+                flag.WaitOne();
+                Assert.Contains(HttpPropertyKeys.DependencyScope, request.Properties);
+                return new HttpResponseMessage(HttpStatusCode.OK);
+            });
+        });
+        var handler = new DependencyScopeHandler { InnerHandler = fakeHandler };
+        var invoker = new HttpMessageInvoker(handler);
+        var task = invoker.SendAsync(request, CancellationToken.None);
+
+        Assert.Contains(HttpPropertyKeys.DependencyScope, request.Properties);
+        flag.Set();
+        task.GetAwaiter().GetResult();
+        Assert.DoesNotContain(HttpPropertyKeys.DependencyScope, request.Properties);
+    }
+
+    [Fact]
+    public void RemoveAutofacDependencyScopeEvenIfTaskThrows()
```

Comment:
Is it worth having one of these tests as async (I think there's an `Assert.ThrowsAsync()`) to check the `ContinueWith` schedules correctly?

Reply:
Sure, I can add that.